integration/ledger/reset_rollback_test.go

```diff
@@ -191,6 +191,57 @@ var _ = Describe("rollback, reset, pause and resume peer node commands", func()
 		Expect(dbPath).To(BeADirectory())
 		helper.assertPresentInCollectionM("marblesp", "marble2", peer)
 	})
+
+	// This test exercises peer node unjoin on the following peers:
+	// Org1.peer0 - unjoin
+	// Org2.peer0 -
+	// Org3.peer0 - unjoin (via partial / resumed deletion on restart)
+	It("unjoins channels and checks side effects on the ledger and transient storage", func() {
+		By("Checking ledger heights on each peer")
+		for _, peer := range helper.peers {
+			Expect(helper.getLedgerHeight(peer)).Should(Equal(14))
+		}
+
+		org1peer0 := setup.network.Peer("Org1", "peer0")
+		org2peer0 := setup.network.Peer("Org2", "peer0")
+		org3peer0 := setup.network.Peer("Org3", "peer0")
+
+		// Negative test: peer node unjoin should fail when the peer is online.
+		By("unjoining the peer while the peer node is online")
+		expectedErrMessage := "as another peer node command is executing," +
+			" wait for that command to complete its execution or terminate it before retrying"
+		helper.unjoin(org1peer0, expectedErrMessage, false)
+		helper.unjoin(org2peer0, expectedErrMessage, false)
+		helper.unjoin(org3peer0, expectedErrMessage, false)
+
+		By("stopping the peers to test unjoin commands")
+		setup.stopPeers()
+
+		By("Unjoining from a channel while the peer is down")
+		helper.unjoin(org1peer0, "", true)
+
+		// Negative test: unjoin when the channel has been unjoined
+		By("Double unjoining from a channel")
+		expectedErrMessage = "unjoin channel \\[testchannel\\]: cannot update ledger status, ledger \\[testchannel\\] does not exist"
+		helper.unjoin(org1peer0, expectedErrMessage, false)
+
+		// Simulate an error in peer unjoin by marking the ledger folder as read-only.
+		By("Unjoining from a peer with a read-only ledger file system")
+		ledgerStoragePath := filepath.Join(setup.network.PeerLedgerDir(org3peer0), "chains/chains", helper.channelID)
+		Expect(os.Chmod(ledgerStoragePath, 0o555)).NotTo(HaveOccurred())
+		Expect(os.RemoveAll(ledgerStoragePath)).To(HaveOccurred()) // can not delete write-only ledger folder
+		expectedErrMessage = "ledgersData/chains/chains/testchannel/blockfile_000000: permission denied"
+		helper.unjoin(org3peer0, expectedErrMessage, false)
+		Expect(os.Chmod(ledgerStoragePath, 0o755)).NotTo(HaveOccurred())
+
+		By("restarting peers")
+		setup.startPeer(org1peer0)
+		setup.startPeer(org2peer0)
+		setup.startPeer(org3peer0)
+
+		helper.assertUnjoinedChannel(org1peer0)
+		helper.assertUnjoinedChannel(org3peer0)
```

Comment:
Would it be easy to add after this -- for making one of these peers to join the channel again?